File: objc/Strategy/Strategy/SadnessAnalyzeStrategy.m
#import "SadnessAnalyzeStrategy.h"

@implementation SadnessAnalyzeStrategy

-(void)analyze:(NSArray<Post*>*)posts {
    int postCount = [posts count];
    int totalSadCount = 0;
    for(Post* post in posts)
        totalSadCount += post.sadCount;
    NSLog(@"Sadness Value: %lf", (double)totalSadCount / postCount);
}

@end

File: swift/Strategy/Strategy/SadnessAnalyzeStrategy.swift
import Foundation

class SadnessAnalysisStrategy : PostAnalyzeStrategy {
    func analyze(_ posts: [Post]) -> Void {
        let postCount: Int = posts.count
        var totalSadCount: Int = 0
        for post in posts {
            totalSadCount += post.sadCount
        }
        let val: Double = Double(totalSadCount / postCount)
        print("Laugh Value: \(val)")
    }
}
